homeassistant/components/hassio/addon_manager.py

```diff
@@ -152,23 +156,23 @@ async def async_get_addon_info(self) -> AddonInfo:
                 version=None,
             )
 
-        addon_info = await async_get_addon_info(self._hass, self.addon_slug)
+        addon_info = await supervisor_client.addons.addon_info(self.addon_slug)
         addon_state = self.async_get_addon_state(addon_info)
         return AddonInfo(
-            available=addon_info["available"],
-            hostname=addon_info["hostname"],
-            options=addon_info["options"],
+            available=addon_info.available,
+            hostname=addon_info.hostname,
+            options=addon_info.options,
             state=addon_state,
-            update_available=addon_info["update_available"],
-            version=addon_info["version"],
+            update_available=addon_info.update_available,
+            version=addon_info.version,
         )
 
     @callback
-    def async_get_addon_state(self, addon_info: dict[str, Any]) -> AddonState:
+    def async_get_addon_state(self, addon_info: InstalledAddonComplete) -> AddonState:
         """Return the current state of the managed add-on."""
         addon_state = AddonState.NOT_RUNNING
 
-        if addon_info["state"] == "started":
+        if addon_info.state == "started":
```

Comment:
I think we know all the states, would it be nice to make this an enum? (can be a followup, just a question)

Reply:
Oh. It is an enum already in the client library. I should've used that, whoops